includes/class-wc-adapter.php

```diff
@@ -0,0 +1,682 @@
+			$shipping = $order->get_address( 'shipping' );
+			$diff     = array_diff( $billing, $shipping );
+			if ( 0 === count( $diff ) ) {
+				// Ship to cardholder's billing address
+				$result[ RiskIndicatorInterface::SHIP_INDICATOR ] = '01';
+			} else {
+				// Ship to address that is different than cardholder's billing address
+				$result[ RiskIndicatorInterface::SHIP_INDICATOR ] = '03';
+			}
+		}
+
+		// @todo Add features of WooThemes Order Delivery and Pre-Orders WooCommerce Extensions
+
+		return apply_filters( 'swedbank_pay_risk_indicator', $result, $order, $this );
+	}
+
+	/**
+	 * Get Payee Info of Order.
+	 *
+	 * @param mixed $order_id
+	 *
+	 * @return array
+	 */
+	public function getPayeeInfo( $order_id ) {
+		$order = wc_get_order( $order_id );
+
+		return array(
+			PayeeInfoInterface::ORDER_REFERENCE => $order->get_id(),
+		);
+	}
+
+	/**
+	 * Update Order Status.
+	 *
+	 * @param mixed $order_id
+	 * @param string $status
+	 * @param string|null $message
+	 * @param mixed|null $transaction_id
+	 */
+	public function updateOrderStatus( $order_id, $status, $message = null, $transaction_id = null ) {
+		$order = wc_get_order( $order_id );
+
+		if ( $order->get_meta( '_payex_payment_state' ) === $status ) {
+			$this->log( LogLevel::WARNING, sprintf( 'Action of Transaction #%s already performed', $transaction_id ) );
+
+			return;
+		}
+
+		if ( $transaction_id ) {
+			$order->update_meta_data( '_transaction_id', $transaction_id );
+			$order->save_meta_data();
+		}
+
+		switch ( $status ) {
+			case OrderInterface::STATUS_PENDING:
+				$order->update_meta_data( '_payex_payment_state', $status );
+				$order->update_status( 'on-hold', $message );
+				break;
+			case OrderInterface::STATUS_AUTHORIZED:
+				$order->update_meta_data( '_payex_payment_state', $status );
+				$order->save_meta_data();
+
+				// Reduce stock
+				$order_stock_reduced = $order->get_meta( '_order_stock_reduced' );
+				if ( ! $order_stock_reduced ) {
+					wc_reduce_stock_levels( $order->get_id() );
+				}
+
+				$order->update_status( 'on-hold', $message );
+
+				break;
+			case OrderInterface::STATUS_CAPTURED:
+				$order->update_meta_data( '_payex_payment_state', $status );
+				$order->save_meta_data();
+
+				$order->payment_complete( $transaction_id );
+				$order->add_order_note( $message );
+				break;
+			case OrderInterface::STATUS_CANCELLED:
+				$order->update_meta_data( '_payex_payment_state', $status );
+				$order->save_meta_data();
+
+				if ( ! $order->has_status( 'cancelled' ) ) {
+					$order->update_status( 'cancelled', $message );
+				} else {
+					$order->add_order_note( $message );
+				}
+				break;
+			case OrderInterface::STATUS_REFUNDED:
+				// @todo Implement Refunds creation
+				// @see wc_create_refund()
+
+				$order->update_meta_data( '_payex_payment_state', $status );
+				$order->save_meta_data();
+
+				if ( ! $order->has_status( 'refunded' ) ) {
+					$order->update_status( 'refunded', $message );
+				} else {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method updateOrderStatus uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)